src/index.ts

```diff
@@ -30,9 +30,17 @@ export class EIP7412 {
 
   async enableERC7412(
     client: viem.PublicClient,
-    tx: TransactionRequest
+    tx: TransactionRequest | TransactionRequest[]
   ): Promise<TransactionRequest> {
-    let multicallCalls: TransactionRequest[] = [tx];
+    let multicallCalls: TransactionRequest[];
+
+    if (Array.isArray(tx)) {
+      multicallCalls = tx;
+    } else {
+      multicallCalls = [tx];
+    }
+
+    const initialMulticallLength = multicallCalls.length;
```

Comment:
wouldn't it be easier to just have two arrays, one with the original list of calls that was requested, and a second which has the prepended resolved oracle data. Then you can just concatenate them together to simulate/return, and remove a ton of math here.

Reply:
I like that. `let multicallCalls = [tx].flat()` You can also do this to create `TransactionRequest[]`